### DISCUSSION TITLE
Listen for events for a specific entity id

### DISCUSSION THREAD

**Discussion by ChristopheCVB** [strapi-community/plugin-io#93] [Q&A]
Hi there!

Firstly, thank you for the good work you've put here, v2 seems to be way cleaner than v1, bravo!

I come here to know how to properly implement events for specific entities (based on the content-type and its id). Of course I'm only talking about something like `{uid}:update:{id}` and `{uid}:delete:{id}`, not the `create`.

Also, if you think this is something worth having by default in the plugin, I'm willing to open a PR, I just wanted to discuss implementation first.

Cheers!

**Comment by ComfortablyCoding** [comment]
Hello and thanks!

It's an interesting question, at some point I did consider. I reasoned it would be too many events to be emitting and left it out. 

I would say the options are as follows:
1. Add it by default to core
2. Add a before/after emit hook to core
3. Add a re-emit service that listens for the top level emit and re-emits it as a single entity one
4. Use the current emit and check the id in the data.

Personally option 2 is probably what I would go with as it allows any custom extension by the user instead of a specific implementation for a given use-case.

P.S.
Unfortunately I am no longer maintaining this plugin (or any of my other plugins) anymore. Due to that I won't be able to help beyond this. 

I have moved them to strapi community so others can maintain them if desired.
